src/registries/npm-registry.js

```diff
@@ -72,9 +73,39 @@ export default class NpmRegistry extends Registry {
     return name.replace('/', '%2f');
   }
 
+  getRequestUrl(registry: string, pathname: string): string {
+    const isUrl = /^https?:/.test(pathname);
+
+    if (isUrl) {
+      return pathname;
+    } else {
+      return url.resolve(registry, pathname);
+    }
+  }
+
+  isRequestToRegistry(requestUrl: string, registryUrl: string): boolean {
+    const normalizedRequestUrl = normalizeUrl(requestUrl);
+    const normalizedRegistryUrl = normalizeUrl(registryUrl);
+    const requestParsed = url.parse(normalizedRequestUrl);
```

Comment:
Simplification:
```
const {requestHost, requestPath} = url.parse(normalizeUrl(requestUrl));
const {registryHost, registryPath} = url.parse(normalizeUrl(registryUrl));
```

Also, `registryUrl` should not change frequently so we may consider memoizing parsing that in a future patch.